### PR TITLE
New Feature - Get Avatar

### PR DESCRIPTION
**Pull request by not-archiey**
This feature will enable the user to get someone's avatar by using a command

Usage : `!avatar @not-archiey`

image

### CODE REVIEW

**Comment by iamdeja** [comment]
That's a great looking PR, I'm excited

File: commands/Avatar.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  constructor() {
+    super({
+      name: "avatar",
+      category: "tool",
+      description: "Returns aforementioned avatar"
+    });
+  }
+
+  async run(client, message) {
+    const user = message.mentions.users.first();
```
Comment (duncte123) [inline commands/Avatar.js]: If you merge develop in your branch you can use finderUtils to find users by name

File: commands/Avatar.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+export default class Avatar extends Command {
+  constructor() {
+    super({
+      name: "avatar",
+      category: "tool",
+      description: "Returns aforementioned avatar"
+    });
+  }
+
+  async run(client, message, args) {
```
Comment (duncte123) [inline commands/Avatar.js]: No need for this to be Async

File: commands/Avatar.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    });
+  }
+
+  run(client, message, args) {
+    const [user] = findUsers(args[0], client);
+    let headerMessage = "Here it is!\n";
+
+    if (!user) {
+      message.channel.send(
+        "Sorry, but you must tag a user to use this command~!"
```
Comment (duncte123) [inline commands/Avatar.js]: Maybe you could send `Sorry but I couldn't find ${args[0]}` also check what happens when you do `!avatar` without arguments

File: commands/Avatar.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import Command from "./base/Command";
+import { findMembers, findUsers } from "../utils/finderUtil";
+import { prefix } from "../yuuto";
```
Comment (duncte123) [inline commands/Avatar.js]: Does yuuto export this?

File: commands/Avatar.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    const [member] =
+      findMembers(args.join(" "), message.guild) ||
+      findUsers(args.join(" "), client);
+
+    if (!member) {
+      message.channel.send(`Sorry, but i couldn't find ${args.join(" ")}`);
+      return;
+    }
```
Comment (duncte123) [inline commands/Avatar.js]: maybe you could default this to message.author

File: commands/Avatar.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    if (!args.length) {
+      message.channel.send(
+        `You can use this command by using it like \`${prefix}avatar @user\``
+      );
+      return;
+    }
+
+    const [member] =
+      findMembers(args.join(" "), message.guild) ||
+      findUsers(args.join(" "), client);
```
Comment (duncte123) [inline commands/Avatar.js]: This probably doesn't work as expected, also store `args.join(" ")` in a varable

File: commands/Avatar.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+      return;
+    }
+
+    const messageArgs = args.join(" ");
+    
+    const [member] =
+      findMembers(messageArgs, message.guild) ||
+      findUsers(messageArgs, client);
+
+    if (!member) {
```
Comment (duncte123) [inline commands/Avatar.js]: I actually meant defaulting the member variable to message.author, I should've been more clear about this.
By defaulting it to message.author you don't have to do an args check anymore and can make the argument optional

File: commands/Avatar.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  run(client, message, args) {
+    // Joins the whole argument onto spaces
+    const messageArgs = args.join(" ");
+    let member;
+
+    if (args.length) {
+      // Uses the user query
+      [member] =
+        findMembers(messageArgs, message.guild) ||
+        findUsers(messageArgs, client);
```
Comment (duncte123) [inline commands/Avatar.js:22]: this won't work as intended, arrays are always true
image
You can however just grab the first element and check on that, javascript won't throw 
image

```suggestion
      member =
        findMembers(messageArgs, message.guild)[0] ||
        findUsers(messageArgs, client)[0];
```

File: commands/Avatar.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    } else {
+      // Uses the sender avatar instead
+      member = message.author;
+    }
+
+    // If query not found
+    if (!member) {
+      message.channel.send(
+        `Sorry, but i couldn't find that... ${message.author}`
```
Comment (duncte123) [inline commands/Avatar.js:31]: The "I" is always a capital ;)
That mention doesn't really make sense to me
```suggestion
        `Sorry, but I couldn't find that...`
```

File: commands/Avatar.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
+import Command from "./base/Command.js";
+import { findMembers, findUsers } from "../utils/finderUtil.js";
+
+export default class Avatar extends Command {
+  constructor() {
+    super({
+      name: "avatar",
+      category: "tool",
```
Comment (duncte123) [inline commands/Avatar.js:8]: I'd put it in `util` instead

File: commands/Avatar.js (hunk 9 above)
Comment (not-archiey) [reply commands/Avatar.js:31]: Sorry for my terrible grammar, I'll resolve this

File: commands/Avatar.js (hunk 8 above)
Comment (not-archiey) [reply commands/Avatar.js:22]: On my way to fix this~!

File: commands/Avatar.js (hunk 10 above)
Comment (not-archiey) [reply commands/Avatar.js:8]: Sure

**Pull request merged**